File: src/main/java/dev/oribuin/skyblock/command/impl/InviteCommand.java
package dev.oribuin.skyblock.command.impl;

import dev.rosewood.rosegarden.RosePlugin;
import dev.rosewood.rosegarden.command.framework.ArgumentsDefinition;
import dev.rosewood.rosegarden.command.framework.BaseRoseCommand;
import dev.rosewood.rosegarden.command.framework.CommandInfo;
import dev.oribuin.skyblock.command.impl.sub.invite.AcceptCommand;
import dev.oribuin.skyblock.command.impl.sub.invite.DenyCommand;
import dev.oribuin.skyblock.command.impl.sub.invite.SendCommand;

public class InviteCommand extends BaseRoseCommand {

    public InviteCommand(RosePlugin rosePlugin) {
        super(rosePlugin);
    }

    @Override
    protected CommandInfo createCommandInfo() {
        return CommandInfo.builder("invite")
                .permission("skyblock.invite")
                .playerOnly(false)
                .arguments(this.createArgumentsDefinition())
                .build();
    }

    protected ArgumentsDefinition createArgumentsDefinition() {
        return ArgumentsDefinition.builder().requiredSub("command",
                new AcceptCommand(this.rosePlugin),
                new DenyCommand(this.rosePlugin),
                new SendCommand(this.rosePlugin)
        );
    }

}

File: Plugin/src/main/kotlin/xyz/oribuin/skyblock/command/command/InviteCommand.kt
package xyz.oribuin.skyblock.command.command

import dev.rosewood.rosegarden.RosePlugin
import dev.rosewood.rosegarden.command.framework.CommandContext
import dev.rosewood.rosegarden.command.framework.RoseCommand
import dev.rosewood.rosegarden.command.framework.RoseCommandWrapper
import dev.rosewood.rosegarden.command.framework.RoseSubCommand
import dev.rosewood.rosegarden.command.framework.annotation.RoseExecutable
import xyz.oribuin.skyblock.command.command.sub.invite.AcceptCommand
import xyz.oribuin.skyblock.command.command.sub.invite.DenyCommand
import xyz.oribuin.skyblock.command.command.sub.invite.SendCommand

@Suppress("unused")
class InviteCommand(rosePlugin: RosePlugin, parent: RoseCommandWrapper) : RoseCommand(rosePlugin, parent,
    AcceptCommand::class.java,
    DenyCommand::class.java,
    SendCommand::class.java
) {

    @RoseExecutable
    fun execute(context: CommandContext, subcommand: RoseSubCommand) {
        // Unused
    }

    override fun getDefaultName(): String = "invite"

    override fun getDescriptionKey(): String = "command-invite-description"

    override fun getRequiredPermission(): String = "skyblock.command.invite"

    override fun isPlayerOnly(): Boolean = true

}